google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITKmsTest.java

```diff
@@ -0,0 +1,537 @@
+import io.grpc.ClientInterceptor;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.auth.MoreCallCredentials;
+import io.grpc.stub.MetadataUtils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.Key;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.crypto.spec.SecretKeySpec;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class ITKmsTest {
+
+  private static String kmsKeyOneResourcePath;
+  private static String kmsKeyTwoResourcePath;
+  private static final String KMS_KEY_RING_NAME = "gcs_test_kms_key_ring";
+  private static final String KMS_KEY_RING_LOCATION = "us";
+  private static final String KMS_KEY_ONE_NAME = "gcs_kms_key_one";
+  private static final String KMS_KEY_TWO_NAME = "gcs_kms_key_two";
+  private static Metadata requestParamsHeader = new Metadata();
+  private static Metadata.Key<String> requestParamsKey =
+      Metadata.Key.of("x-goog-request-params", Metadata.ASCII_STRING_MARSHALLER);
+  private static ManagedChannel kmsChannel;
+  @ClassRule public static final StorageFixture storageFixture = StorageFixture.defaultHttp();
+  private static Storage storage;
+  private static final Logger log = Logger.getLogger(ITKmsTest.class.getName());
+  // Probably replace this with bucket fixture.
+  private static final String BUCKET = RemoteStorageHelper.generateBucketName();
+  private static final byte[] BLOB_BYTE_CONTENT = {0xD, 0xE, 0xA, 0xD};
+  private static final String BLOB_STRING_CONTENT = "Hello Google Cloud Storage!";
+  private static final String BASE64_KEY = "JVzfVl8NLD9FjedFuStegjRfES5ll5zc59CIXw572OA=";
+  private static final Key KEY =
+      new SecretKeySpec(BaseEncoding.base64().decode(BASE64_KEY), "AES256");
+  private static final String CONTENT_TYPE = "text/plain";
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    // Prepare KMS KeyRing for CMEK tests
+    prepareKmsKeys();
+    storage.create(BucketInfo.newBuilder(BUCKET).build());
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (kmsChannel != null) {
+      try {
+        kmsChannel.shutdownNow();
+      } catch (Exception e) {
+        log.log(Level.WARNING, "Error while trying to shutdown kms channel", e);
+      }
+      kmsChannel = null;
+    }
+  }
+
+  private static void prepareKmsKeys() throws IOException {
+    storage = storageFixture.getInstance();
+    // https://cloud.google.com/storage/docs/encryption/using-customer-managed-keys
+    String projectId = storage.getOptions().getProjectId();
+    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
```

Comment:
Better to use the credentials from storage
```suggestion
    Credentials credentials = storage.getOptions().getCredentials();
```